sdk/highlight-next/src/util/with-highlight-config.ts

```diff
@@ -137,10 +137,10 @@ const getHighlightConfig = async (
 				source: '/:path*.map',
 				destination: '/404',
 			}
-
+			const backendUrl = defaultOpts.sourceMapsBackendUrl
 			const highlightRewrite = {
 				source: '/highlight-events',
-				destination: 'https://pub.highlight.io',
+				destination: backendUrl || 'https://pub.highlight.io',
```

Comment:
The change required in the ticket is a bit different. Essentially, as part of the `withHighlightConfig` rewrite, we want to make it so the `<HighlightInit />` component automatically updates the client-side `H.init` [call](https://github.com/highlight/highlight/blob/e6a5bcea42b986a4da2fecd5234e18bf0758a39a/sdk/highlight-next/src/next-client.tsx#L26-L29) with the `backendUrl`

[Here's another full example app](https://github.com/highlight/highlight/tree/main/e2e/nextjs) if that helps.